bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/radiora/RS232Connection.java

```diff
@@ -59,7 +63,8 @@ public void open(String portName, int baud) throws RadioRAConnectionException {
         }
 
         try {
-            serialPort = portIdentifier.open("openhab", 5000);
+            SerialPort serialPort = portIdentifier.open("openhab", 5000);
```

Comment:
Not really necessary to create a local variable here

Reply:
Yes. I agree with you, but wherever you see this pattern it was done to eliminate compiler warnings associated with the Eclipse JDT null annotations. It seems to "helpfully" assume that if something is annotated as being Nullable but isn't a local variable, another thread may have come along and changed it to null behind your back. No matter if code design or use of locking prevents this. So the solution is to use a lot of otherwise unnecessary local variable. It would be interesting to see what the overall impact on GC is.